File: Homework2CS/Homework2_1CS/Homework2_1CS/Form1.cs
namespace Homework2_1CS
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }
        int press = 0;
        private void button1_Click(object sender, EventArgs e)
        {
            if (press == 0)
            {
                press = 1;
                this.button1.Text = "Pause";
                this.timer1.Start();
            }
            else
            {
                press = 0;
                this.button1.Text = "Play";
                this.timer1.Stop();
            }
        }

        Random rand = new Random(); 
        private void timer1_Tick(object sender, EventArgs e)
        {
            this.timer1.Interval = rand.Next(1, 10) * rand.Next(100,300);        
            this.progressBar1.PerformStep();
            if (this.progressBar1.Value == 100)
            {
                this.button1.Visible = false;
            }
        }
    }
}

File: Homework2VB/Homework2_1VB/Homework2_1/Form1.vb
Public Class Form1
    Public rand As New Random
    Private Sub Timer1_Tick(sender As Object, e As EventArgs) Handles Timer1.Tick
        Me.RichTextBox1.AppendText(rand.Next & vbCrLf)
    End Sub

    Dim playpause As Integer = 0
    Private Sub Button1_Click(sender As Object, e As EventArgs) Handles Button1.Click
        If playpause = 0 Then
            Me.Button1.Text = "Pause"
            playpause = 1
            Timer1.Start()
        Else
            Me.Button1.Text = "Play"
            playpause = 0
            Timer1.Stop()
        End If

    End Sub
End Class
